Scripts/Runtime/Entities/Tasks/AbstractTaskDriverSystem.cs

```diff
@@ -0,0 +1,134 @@
+        /// <see cref="ConfigureUpdateJob"/>
+        /// </summary>
+        protected abstract void InitUpdateJobConfiguration();
+        
+        protected override void OnDestroy()
+        {
+            m_InstanceData.Dispose();
+            
+            m_UpdateJobData.Clear();
+
+            foreach (AbstractTaskDriver taskDriver in m_TaskDrivers)
+            {
+                taskDriver.Dispose();
+            }
+            m_TaskDrivers.Clear();
+
+            base.OnDestroy();
+        }
+
+        protected JobTaskWorkConfig ConfigureUpdateJob(JobTaskWorkConfig.JobDataDelegate jobDataDelegate)
+        {
+            JobTaskWorkConfig config = new JobTaskWorkConfig(jobDataDelegate, this);
+            m_UpdateJobData.Add(config);
+            return config;
+        }
+        
+        //TODO: #39 - Some way to remove the update Job
+
+        protected VirtualData<TKey, TInstance> CreateData<TKey, TInstance>(params AbstractVirtualData[] sources)
+            where TKey : unmanaged, IEquatable<TKey>
+            where TInstance : unmanaged, IKeyedData<TKey>
+        {
+            VirtualData<TKey, TInstance> virtualData = VirtualData<TKey, TInstance>.Create(sources);
+            m_InstanceData.AddData(virtualData);
+            return virtualData;
+        }
+
+        protected VirtualData<TKey, TInstance> GetData<TKey, TInstance>()
+            where TKey : unmanaged, IEquatable<TKey>
+            where TInstance : unmanaged, IKeyedData<TKey>
+        {
+            return m_InstanceData.GetData<TKey, TInstance>();
+        }
+
+        internal void RegisterTaskDriver(AbstractTaskDriver taskDriver)
+        {
+            m_TaskDrivers.Add(taskDriver);
+        }
+        
+        protected sealed override void OnUpdate()
+        {
+            //Have drivers be given the chance to add to the Instance Data
+            JobHandle driversPopulateHandle = m_PopulateBulkScheduler.BulkSchedule(Dependency);
+            
+            //Consolidate our instance data to operate on it
+            JobHandle consolidateInstancesHandle = m_InstanceData.ConsolidateForFrame(driversPopulateHandle);
+            
+            //TODO: #38 - Allow for cancels to occur
+            
+            //Allow the generic work to happen in the derived class
+            JobHandle updateInstancesHandle = m_JobTaskWorkConfigBulkScheduler.BulkSchedule(consolidateInstancesHandle);
+            
+            //Have drivers consolidate their result data
+            JobHandle driversConsolidateHandle = m_ConsolidateBulkScheduler.BulkSchedule(updateInstancesHandle);
+            
+            //TODO: #38 - Allow for cancels on the drivers to occur
+            
+            //Have drivers to do their own generic work
+            JobHandle driversUpdateHandle = m_UpdateBulkScheduler.BulkSchedule(driversConsolidateHandle);
+
+            //Ensure this system's dependency is written back
+            Dependency = driversUpdateHandle;
```

Comment:
The AbstractAnvilBase check is a debug only cost. This is what helps enforce your sequence and tells you at the moment the dependency diverts rather than having to look through all dependency changes in the sequence.

`CheckedState()` is a minor null check and the `SystemState->Dependency` getter is only expensive on the first get per frame. That boolean flag only gets flipped in `SystemState.BeforeOnUpdate` from what I can tell.

I think I'd agree with you more if this was in a Job or iteration but in release this comes down to an extra null check on some indirection a few hundred times per frame in exchange for greater safety and more modular code where you can easily comment out a line for testing.